src/encoded/static/components/matrix_deeply_profiled.js

```diff
@@ -0,0 +1,576 @@
+import { FacetList, SearchControls } from './search';
+import { matrixDescription, isAllDeeplyMatrix, switchDeeplyProfilePageType, MatrixAddCart } from './matrix_deeply_profiled_uniform_batch';
+
+
+/**
+ * Number of subcategory items to show when subcategory isn't expanded.
+ * @constant
+ */
+const SUB_CATEGORY_SHORT_SIZE = 5;
+
+/** Page name */
+const matrixName = 'deeply-profiled-matrix';
+
+/**
+ * Given one subcategory of matrix data (all the subcategory rows within a category), collect
+ * summary information about that data.
+ * @param {array}  subCategoryData Array of subcategory objects, each containing an array of data
+ * @param {string} columnCategoryType `subCategoryData` property that contains data array
+ * @param {array}  colTitleMap Maps column titles to the column indices they correspond to
+ * @param {number} colCount Number of columns in matrix
+ *
+ * @return {object} Summary information about given matrix data:
+ *     {
+ *         {array} subCategorySums: Column sums for all subcategory rows
+ *         {number} maxSubCategoryValue: Maximum value in all cells in all subcategory rows
+ *         {number} minSubCategoryValue: Minimum value in all cells in all subcategory rows
+ *     }
+ */
+const analyzeSubCategoryData = (subCategoryData, columnCategoryType, colTitleMap, colCount) => {
+    const subCategorySums = Array(colCount).fill(0);
+    let maxSubCategoryValue = 0;
+    let minSubCategoryValue = Number.MAX_VALUE;
+
+    subCategoryData.forEach((rowData) => {
+        // `rowData` has all the data for one row. Collect sums of all data for each column.
+        rowData[columnCategoryType].buckets.forEach((value) => {
+            const colIndex = colTitleMap[value.key];
+            subCategorySums[colIndex] = (subCategorySums[colIndex] || 0) + value.doc_count;
+        });
+
+        // Update min and max values found within all subcategories of the given category.
+        const rowDataValues = rowData[columnCategoryType].buckets.map((bucket) => bucket.doc_count);
+        const prospectiveMax = Math.max(...rowDataValues);
+        if (maxSubCategoryValue < prospectiveMax) {
+            maxSubCategoryValue = prospectiveMax;
+        }
+        const prospectiveMin = Math.min(...rowDataValues.filter((value) => value));
+        if (minSubCategoryValue > prospectiveMin) {
+            minSubCategoryValue = prospectiveMin;
+        }
+    });
+    return { subCategorySums, maxSubCategoryValue, minSubCategoryValue: minSubCategoryValue - 1 };
+};
+
+
+let _navbarHeight = null;
+
+/**
+ * Get height of the navbar
+ *
+ * @returns height of navbar on first call and restores same value until page is garbage collected
+ */
+const getNavbarHeight = () => {
+    if (_navbarHeight === null) {
+        const navbar = document.querySelector('#navbar');
+        _navbarHeight = navbar ? navbar.getBoundingClientRect().height : 0;
+    }
+    return _navbarHeight;
+};
```

Comment:
At some point we should put this function into matrix.js and just import it where needed. We have about five copies of it.